src/franka_plan_runner.cc

```diff
@@ -706,57 +704,60 @@ franka::JointPositions FrankaPlanRunner::JointPositionCallback(
   }
   output_to_franka = utils::EigenToArray(output_to_franka_eigen);
 
-  if (franka_time_ > plan_->end_time()) {
+  if (franka_time_ > plan_->end_time()) {  // check convergence
+    // the following two constants must be tuned together
+    static const double CONV_ANGLE_THRESHOLD {0.001};  // rad, empirical
+    static const double CONV_SPEED_THRESHOLD {0.003};  // rad/s, L2 norm
```

Comment:
Can we define these in a header somewhere, or make it a command line parameter, or YAML parameter?

Reply:
it was in the header. moved here to reduce scope because they are local constants and don't need to be class members.